miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/api/ApiClient.kt

```diff
@@ -101,13 +105,40 @@ internal class RetrofitRequestExecutor(
     private val retrofit: Retrofit
 ) {
 
-    private inline fun <reified T : ErrorResponse> createErrorConvertor(retrofit: Retrofit) =
+    private var retryCount = 0
+
+    private inline fun <reified T : ErrorResponse> createErrorConverter(retrofit: Retrofit) =
         retrofit.responseBodyConverter<T>(T::class.java, arrayOfNulls<Annotation>(0))
 
-    @Suppress("TooGenericExceptionCaught", "ThrowsCount")
+    @Suppress(
+        "TooGenericExceptionCaught",
+        "ThrowsCount",
+        "MagicNumber",
+        "LongMethod",
+        "NestedBlockDepth",
+        "ComplexMethod"
+    )
     suspend fun <T> executeRequest(call: Call<T>): T = try {
         val response = call.execute()
 
+        // retry network request when there is 500 error code from the server
+        if (response.code() >= 500) {
+            if (retryCount++ < TOTAL_RETRIES) {
+                Log.v(MINIAPP_NETWORK_RETRY, "Retrying in $retryCount out of $TOTAL_RETRIES times.")
+
+                // setting retrying policy about waiting time
+                val backOff = 2.0
+                val waitTime = 0.5 * backOff.pow(retryCount.toDouble())
+                try {
+                    Thread.sleep(waitTime.toLong())
```

Comment:
I am not sure, if this is the good solution for now, but raised this PR for a discussion. 
Based on the final conclusion, I'll try to improve the test coverage.

Reply:
Maybe you can use a `delay` before the `executeRequest` call? I'm not a Kotlin expert though

Reply:
Yeah, `delay` is a good idea!